fieldservice_repair_order_template/models/fsm_order.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2024 Camptocamp SA (https://www.camptocamp.com).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class FSMOrder(models.Model):
+    _inherit = "fsm.order"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        # OVERRIDE to apply the repair order template on the related repair order
+        records = super().create(vals_list)
+        records._apply_repair_order_template()
+        return records
+
+    def write(self, vals):
+        # OVERRIDE to apply the repair order template on the related repair order
+        res = super().write(vals)
+        if "template_id" in vals:
+            self._apply_repair_order_template()
+        return res
+
+    def _onchange_template_id(self):
+        res = super()._onchange_template_id()
```

Comment:
it seems onchange methods must be public otherwise they are not triggered **on the UI**
```suggestion
    def onchange_template_id(self):
        res = super().onchange_template_id()
```
cf. https://github.com/OCA/field-service/pull/1449

Reply:
My bad, it was the missing decorator the culprit